packages/blade-mcp/src/createServer.ts

```diff
@@ -74,28 +81,28 @@ export const createServer = (): McpServer => {
     createBladeCursorRulesToolName,
     createBladeCursorRulesToolDescription,
     createBladeCursorRulesToolSchema,
-    createBladeCursorRulesToolCallback,
+    createBladeCursorRulesToolCallback(transportType),
   );
 
   server.tool(
     getBladeComponentDocsToolName,
     getBladeComponentDocsToolDescription,
-    getBladeComponentDocsToolSchema,
-    getBladeComponentDocsToolCallback,
+    getBladeComponentDocsToolSchema(transportType),
+    getBladeComponentDocsToolCallback(transportType),
   );
 
   server.tool(
     getBladePatternDocsToolName,
     getBladePatternDocsToolDescription,
-    getBladePatternDocsToolSchema,
-    getBladePatternDocsToolCallback,
+    getBladePatternDocsToolSchema(transportType),
+    getBladePatternDocsToolCallback(transportType),
   );
 
```

Comment:
```ts
if (transportType === 'http') {
  server.tool(
      getBladePatternDocsToolName,
      getBladePatternDocsToolDescription,
      getBladePatternDocsToolHTTPSchema,
      getBladePatternDocsToolHTTPCallback,
  );
} else {
  server.tool(
      getBladePatternDocsToolName,
      getBladePatternDocsToolDescription,
      getBladePatternDocsToolSTDIOSchema,
      getBladePatternDocsToolSTDIOCallback,
  );
}
```

would this simplify the types and conditioning in the other files? Right now we have one function here which calls two functions which call one core function again so a lot of chaining of function calls 😅